### DISCUSSION TITLE
How to configure `options = { border = "top" }` on mini.indentscope only for Python?

### DISCUSSION THREAD

**Discussion by robsonpeixoto** [LazyVim/LazyVim#559] [Q&A] [answered]
By default `mini.indentscope` use `options = { border = "both" } ` configuration, but IMHO, `border = "top" ` work better for Python.

Any tip of how to change it only for Python?

**Comment by dpetka2001** [comment]
You could try setting an autocommand for Python files with the following `vim.b.miniindentscope_config = { border = 'top' }` like mentioned [here](https://github.com/echasnovski/mini.nvim/blob/d42d1e5b5394f1a34763dc630f70dffe2470ef84/doc/mini.txt#L275). Haven't tested it, so perhaps you might need some more tweaks.

**Reply by robsonpeixoto** [reply]
**WARN:** Do not use this solution!

Thanks! I put it in my `python.lua` plugin.

```lua
return {
-- ...
  {
    "echasnovski/mini.indentscope",
    init = function()
      vim.api.nvim_create_autocmd("FileType", {
        pattern = { "python" },
        callback = function()
          vim.b.miniindentscope_config = { options = { border = "top" } }
        end,
      })
    end,
  },

-- ...
}
```

**Reply by dpetka2001** [reply]
Does it work with `pattern = {"python"}`? I was under the impression that autocmd patterns had to be `*.py` for example for python files.

**Reply by robsonpeixoto** [reply, marked as answer]
The solution above is WRONG! The `init = function()` will replace the default `init` configuration!



You just need to add the code below on file `lua/config/autocmds.lua`:
```lua
vim.api.nvim_create_autocmd("FileType", {
  pattern = { "make", "python" },
  callback = function()
    vim.b.miniindentscope_config = { options = { border = "top" } }
  end,
  desc = "configure mini.indentscope for languages that use indentation to define blocks",
})
```
